osu.Game/Database/LegacyExporter.cs

```diff
@@ -37,8 +39,15 @@ protected LegacyExporter(Storage storage)
         /// <param name="item">The item to export.</param>
         public void Export(TModel item)
         {
-            string filename = $"{item.GetDisplayString().GetValidFilename()}{FileExtension}";
+            var itemFilename = item.GetDisplayString().GetValidFilename();
 
+            var existingExports = exportStorage.GetFiles("", $"{itemFilename}*{FileExtension}").ToArray();
```

Comment:
You can use LINQ `.Select(f => ...)` instead of `.ToArray()` to avoid array allocation.